holidays/countries/turkmenistan.py

```diff
@@ -0,0 +1,153 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomIslamicHolidays
+from holidays.calendars.gregorian import MAR, APR, MAY, JUN, JUL, AUG, SEP, OCT, NOV
+from holidays.groups import InternationalHolidays, IslamicHolidays
+from holidays.observed_holiday_base import ObservedHolidayBase, SUN_TO_NEXT_WORKDAY
+
+
+class Turkmenistan(ObservedHolidayBase, InternationalHolidays, IslamicHolidays):
+    """Turkmenistan holidays.
+
+    References:
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Turkmenistan>
+        * [Labor Code](https://web.archive.org/web/20250419204037/https://www.tds.gov.tm/en/20-labor-code-of-turkmenistan)
+        * [State Flag and Constitution Day](https://en.wikipedia.org/wiki/State_Flag_and_Constitution_Day_(Turkmenistan))
+        * [Independence Day](https://en.wikipedia.org/wiki/Independence_Day_(Turkmenistan))
+        * [Detailed Research](https://archive.org/details/holiday-research-in-central-asia)
```

Comment:
```suggestion
        * [Detailed Research](https://archive.org/details/holiday-research-in-central-asia)
        * [National Holidays of Turkmenistan (2007-2012)](https://web.archive.org/web/20071024010126/http://www.turkmenistan.gov.tm/_ru/holiday/)
        * [Law on Amendments to the Labor Code, October 9th, 2017](https://web.archive.org/web/20180106073806/http://www.infoabad.com/novosti-turkmenistana/v-turkmenistane-utverzhd-n-novyi-perechen-prazdnichnyh-i-pamjatnyh-dat.html)
```
AFAIK "Professional Holidays" tab on the English version of the Wikipedia page is the `WORKDAY` category according to the 2017 amendment above

Reply:
I had discussed this with @KJhellico in Slack's general channel and he said to skip those.

Reply:
Let's leave them then, I guess :grinning:

Reply:
These changes shouldn't be done then? (The suggestion you made above about references)